DNN Platform/DotNetNuke.Web/InternalServices/EventLogServiceController.cs

```diff
@@ -33,17 +33,16 @@ public HttpResponseMessage GetLogDetails(string guid)
 
             try
             {
-                var logInfo = new LogInfo { LogGUID = guid };
-                logInfo = EventLogController.Instance.GetSingleLog(logInfo, LoggingProvider.ReturnType.LogInfoObjects) as LogInfo;
-                if (logInfo == null)
+                var log = EventLogController.Instance.GetLog(guid);
+                if (log == null || !(log is LogInfo))
```

Comment:
```suggestion
                if (!(EventLogController.Instance.GetLog(guid) is LogInfo log))
```

Reply:
😕 I'm not sure if I agree with this recommendation
- Isn't this doing exactly the same thing?
- Is this a faster computation?

Reply:
It is doing exactly the same thing (`is` pattern matching won't match `null` for a type pattern), and it is faster b/c it's storing the cast instead of doing the cast a second time.

Reply:
That makes sense, I much rather prefer the 2 statements and remove the null check then. I find it easier to read if we declare the `var log` on a separate line from the if statement.
```suggestion
                var log = EventLogController.Instance.GetLog(guid);
                if (!(log is LogInfo))
```

Is this a good compromise?

Reply:
```suggestion
                var log = EventLogController.Instance.GetLog(guid);
                if (!(log is LogInfo logInfo))
```

The main missing piece is adding a variable to store the `LogInfo` instance after casting.  This will let us remove the cast later on.